File: src/content/algorithms/insertion-sort/code.py
def insertion_sort(array):  # #@function-def
    for i in range(1, len(array)):  # #@outer-loop
        key = array[i]  # #@key
        j = i - 1

        while j >= 0 and array[j] > key:  # #@inner-loop
            array[j + 1] = array[j]  # #@shift
            j -= 1

        array[j + 1] = key  # #@insert

    return array

File: src/content/algorithms/insertion-sort/code.rs
pub fn insertion_sort<T: Ord>(arr: &mut [T]) {  // #@generic-function
    let n = arr.len();

    for i in 1..n {  // #@outer-loop
        let mut j = i;  // #@key

        while j > 0 && arr[j - 1] > arr[j] {  // #@while-loop
            arr.swap(j - 1, j);  // #@shift
            j -= 1;
        }
    }
}
